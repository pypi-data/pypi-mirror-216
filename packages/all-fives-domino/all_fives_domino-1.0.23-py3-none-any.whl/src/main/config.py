TARGET_SCORE = 305