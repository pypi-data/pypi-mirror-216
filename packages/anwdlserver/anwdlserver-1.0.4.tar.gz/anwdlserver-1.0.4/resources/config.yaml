# Configuration file for Anweddol server
#
# This document contains all of the configurations that the server 
# will be using. Comment or un-comment variables to change their 
# default values. Refer to the units explanations below, 
# and the official anweddol server documentation.

# ---
# Paths specifications for nessessary files
paths:

  # Server log file path 
  log_file_path: /var/log/anweddol/runtime.txt

  # PID file path for server daemon process
  pid_file_path: /etc/anweddol/daemon_process.pid

  # Live OS image path that will be used by containers 
  container_iso_path: /etc/anweddol/iso/anweddol_container.iso

  # RSA keys root path
  rsa_keys_root_path: /etc/anweddol/rsa_keys

  # Access token database file path
  access_tokens_database_path: /etc/anweddol/credentials/access_token.db

# ---
# Contains all nessessary values for container management
container:

  # Max amount of containers that can run at the same time
  # Zero is not allowed
  max_allowed_running_containers: 6

  # Memory allocated to a container, exprimed in MB
  container_memory: 1024

  # Numbers of virtual CPUs that are allocated to a container
  container_vcpus: 2

  # NAT interface name that containers will be using to receive
  # server administration. Default is 'virbr0', which is the default 
  # interface created by the libvirtd daemon when launched.
  nat_interface_name: virbr0

  # Bridged interface name that containers will be using to
  # communicate with clients. Default is 'anwdlbr0', which is 
  # an universal interface name. 
  # You can use another interface as long as it permits 
  # containers and clients to communicate.
  bridge_interface_name: anwdlbr0

  # Container endpoint username / password 
  # NOTE : The container endpoint default credentials are public and
  # normalized, that's why they are stored here unencrypted. 
  endpoint_username: endpoint
  endpoint_password: endpoint
  endpoint_listen_port: 22

# ---
# Nessessary parameters for server listen interface
server:

  # Privilege-separated user with which to run the server
  # It is not recommended to launch the server with root privileges
  user: anweddol

  # Server listen informations
  bind_address: 0.0.0.0
  listen_port: 6150
  timeout: 10

  # Generate RSA key pair on start and ignore the stored one
  # May increase startup time
  enable_onetime_rsa_keys: False

# ---
# IP filtering parameters
ip_filter:

  # Enable this feature or not
  # If this value is set to 'false', all parameters below will be ignored 
  enabled: False

  # List of allowed / denied IPs
  # If 'any' is specified in the denied_ip_list list, the server will search in
  # the allowed_ip_list list for the IP in question. If specified, the server 
  # will accept the connection, and will deny it otherwise
  denied_ip_list: []

  allowed_ip_list:
    - any

# ---
# Access token parameters
access_token:

  # Enable this feature or not
  # If this value is set to 'false', all parameters below will be ignored 
  enabled: False
