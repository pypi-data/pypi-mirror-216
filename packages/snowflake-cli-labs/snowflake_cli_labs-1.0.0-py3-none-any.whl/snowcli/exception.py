from click.exceptions import ClickException


class EnvironmentVariableNotFoundError(ClickException):
    def __init__(self, env_variable_name: str):
        super().__init__(f"Environment variable {env_variable_name} not found")


class MissingConfiguration(ClickException):
    pass
