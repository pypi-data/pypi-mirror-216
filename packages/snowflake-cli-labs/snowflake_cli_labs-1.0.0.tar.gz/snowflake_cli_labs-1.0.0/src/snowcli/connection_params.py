from dataclasses import dataclass


@dataclass
class ConnectionParams:
    user: str
    password: str
    account: str
    host: str = "snowflakecomputing.com"
    port: int = 443
    protocol: str = "https"
    # add a new parameter to support registry login
    session_parameters: str = '{"PYTHON_CONNECTOR_QUERY_RESULT_FORMAT": "json"}'
