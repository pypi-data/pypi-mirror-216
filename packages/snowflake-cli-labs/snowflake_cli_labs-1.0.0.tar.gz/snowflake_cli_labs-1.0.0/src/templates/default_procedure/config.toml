# The values in this file will be used if flags aren't given
# in CLI commands. Can create multiple --environments (default 'dev')

[connections]
[connections.dev]
database = "<DATABASE_NAME>"
schema = "<SCHEMA_NAME>"
warehouse = "<WAREHOUSE_NAME>"
role = "<ROLE_NAME>"
