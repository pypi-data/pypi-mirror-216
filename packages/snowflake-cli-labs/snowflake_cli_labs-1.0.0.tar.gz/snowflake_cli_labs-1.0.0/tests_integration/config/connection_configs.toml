[connections.integration-tests]
accountname = $SNOWCLI_LOCAL_ACCOUNT_NAME
username = $SNOWCLI_LOCAL_USER
password = $SNOWCLI_LOCAL_PASS
host = $SNOWCLI_LOCAL_HOST
port = $SNOWCLI_LOCAL_PORT
protocol = $SNOWCLI_LOCAL_PROTOCOL
