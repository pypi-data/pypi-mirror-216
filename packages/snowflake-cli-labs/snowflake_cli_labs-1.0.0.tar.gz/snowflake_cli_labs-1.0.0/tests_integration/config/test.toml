snowsql_config_path = "connection_configs.toml"
snowsql_connection_name = "integration-tests"

[dev]
database = "test"
schema = "public"
role = "accountadmin"
warehouse = "regress"
