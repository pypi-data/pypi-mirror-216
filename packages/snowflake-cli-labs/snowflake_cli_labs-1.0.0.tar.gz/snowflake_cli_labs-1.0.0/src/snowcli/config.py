from __future__ import annotations

import os
from pathlib import Path
from typing import Optional, Any

import tomlkit
from tomlkit import dump, table, TOMLDocument
from tomlkit.exceptions import NonExistentKey
import logging

from snowcli.exception import MissingConfiguration
from snowcli.snow_connector import SnowflakeConnector
from snowflake.connector.constants import CONFIG_FILE
from snowflake.connector.config_manager import ConfigManager


log = logging.getLogger(__name__)
snowflake_connection: SnowflakeConnector


class CliConfigManager(ConfigManager):
    def __init__(self, file_path: Path = CONFIG_FILE):
        super().__init__(name="SNOWCLI_PARSER", file_path=file_path)
        self._add_options()

    def from_context(self, config_path_override: Optional[Path]):
        if config_path_override:
            self.file_path = config_path_override
        if not self.file_path.exists():
            self._initialise_config()
        self.read_config()

    def _add_options(self):
        self.add_option(
            name="options",
            parse_str=tomlkit.parse,
        )
        self.add_option(
            name="connections",
            parse_str=tomlkit.parse,
        )

    def get_section(self, *path) -> TOMLDocument:
        section = self
        idx = 0
        while idx < len(path):
            section = section[path[idx]]
            idx += 1
        return section

    def section_exists(self, *path) -> bool:
        try:
            self.get_section(*path)
            return True
        except NonExistentKey:
            return False

    def _get_from_env(self, *path, key: str):
        env_variable_name = (
            "SNOWFLAKE_" + "_".join(p.upper() for p in path) + f"_{key.upper()}"
        )
        return os.environ.get(env_variable_name)

    def get(self, *path, key: str, default: Optional[Any] = None) -> Any:
        """Looks for given key under nested path in toml file."""
        env_variable = self._get_from_env(*path, key=key)
        if env_variable:
            return env_variable
        try:
            return self.get_section(*path)[key]
        except NonExistentKey:
            if default:
                return default
            raise

    def initialize_connection_section(self):
        self.conf_file_cache = TOMLDocument()
        self.conf_file_cache.append("connections", table())

    def _initialise_config(self):
        os.makedirs(os.path.dirname(self.file_path), exist_ok=True)
        self.initialize_connection_section()
        self._dump_config()
        log.info(f"Created Snowflake configuration file at {cli_config.file_path}")

    def get_connection(self, connection_name: str) -> dict:
        try:
            return self.get("connections", key=connection_name)
        except NonExistentKey:
            raise MissingConfiguration(
                f"Connection {connection_name} is not configured"
            )

    def add_connection(self, name: str, parameters: dict):
        if not self.section_exists("connections"):
            self.initialize_connection_section()
        self.get_section("connections").add(name, parameters)
        self._dump_config()

    def _dump_config(self):
        with open(self.file_path, "w+") as fh:
            dump(self.conf_file_cache, fh)


def config_init(config_file: Path):
    """
    Initializes the app configuration. Config provided via cli flag takes precedence.
    If config file does not exist we create an empty one.
    """
    cli_config.from_context(config_path_override=config_file)


def connect_to_snowflake(connection_name: Optional[str] = None, **overrides):  # type: ignore
    connection_name = connection_name if connection_name is not None else "dev"
    return SnowflakeConnector(
        connection_parameters=cli_config.get_connection(connection_name),
        overrides=overrides,
    )


def is_auth():
    # To be removed. Added to simplify refactor
    return True


cli_config: CliConfigManager = CliConfigManager()  # type: ignore
