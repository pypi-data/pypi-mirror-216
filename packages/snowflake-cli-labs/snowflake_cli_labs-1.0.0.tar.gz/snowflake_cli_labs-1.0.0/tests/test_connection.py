import os
from tempfile import NamedTemporaryFile
from textwrap import dedent
from unittest import mock

import pytest

from snowcli.config import CliConfigManager


def test_new_connection_can_be_added(runner, snapshot):
    with NamedTemporaryFile("w+", suffix=".toml") as tmp_file:
        result = runner.invoke(
            [
                "--config-file",
                tmp_file.name,
                "connection",
                "add",
                "--connection-name",
                "conn1",
                "--username",
                "user1",
                "--password",
                "password1",
                "--account",
                "account1",
            ]
        )
        content = tmp_file.read()
    assert result.exit_code == 0, result.output
    assert content == snapshot


def test_fails_if_existing_connection(runner):
    with NamedTemporaryFile("w+", suffix=".toml") as tmp_file:
        tmp_file.write(
            dedent(
                """\
        [connections]
        [connections.conn2]
        username = "foo"
        """
            )
        )
        tmp_file.flush()
        result = runner.invoke(
            [
                "--config-file",
                tmp_file.name,
                "connection",
                "add",
                "--connection-name",
                "conn2",
                "--username",
                "user1",
                "--password",
                "password1",
                "--account",
                "account1",
            ]
        )
    assert result.exit_code == 1, result.output
    assert "Connection conn2 already exists  " in result.output
