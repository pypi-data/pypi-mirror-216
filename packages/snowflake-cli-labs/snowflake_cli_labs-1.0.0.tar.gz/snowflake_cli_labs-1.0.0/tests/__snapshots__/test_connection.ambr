# serializer version: 1
# name: test_new_connection_can_be_added
  '''
  [connections.conn1]
  account = "account1"
  user = "user1"
  password = "password1"
  
  '''
# ---
